src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/rest/TokenCrumbExclusion.java

```diff
@@ -0,0 +1,29 @@
+package com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.rest;
+
+import hudson.Extension;
+import hudson.security.csrf.CrumbExclusion;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Extension
+public class TokenCrumbExclusion extends CrumbExclusion {
+
+    static final String EXCLUSION_PATH = "/bitbucket/oauth/";
```

Comment:
I wanted to define this in terms of constants, but the path is generated by stapler, so I've hard-coded it here